src/clib/lib/enkf/enkf_obs.cpp

```diff
@@ -212,8 +212,8 @@ void enkf_obs_add_obs_vector(enkf_obs_type *enkf_obs,
     if (vector != NULL) {
         const char *obs_key = obs_vector_get_key(vector);
         if (hash_has_key(enkf_obs->obs_hash, obs_key))
-            util_abort("%s: Observation with key:%s already added.\n", __func__,
-                       obs_key);
+            throw std::runtime_error(
+                fmt::format("Duplicate observation: {}", obs_key));
```

Comment:
We provide a convenience exception class for this:

```suggestion
            throw exc::runtime_error("Duplicate observation: {}", obs_key);
```

I think it's in `#include <ert/except.hpp>` if not already included.